### PR TITLE
Add solution for publishing Formulas from private repos using goreleaser

### PR DESCRIPTION
**Pull request by glooms**
Adds a `publish.go` file which checks for any new releases from the private qlik-cli repository, pulls down the relevant assets for those releases, and updates the formula `qlik-cli.rb` so that its download links points to this repository instead of the private one.

Since the script has been run, it also adds all the releases and a formula.

~~**TODO**: Fix some sort of web-hook so that we can run `publish.go` from the other publishing pipeline. My suggestion is a CircleCI step.~~

Resort to manual process for patching the releases. The alternative is to make commits to master from a CCI job. Or something completely different.

### CODE REVIEW

File: Formula/qlik-cli.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# This file was generated by GoReleaser. DO NOT EDIT.
+class QlikCli < Formula
+  desc "qlik-cli is a CLI for using Qlik Sense Cloud editions."
+  homepage "https://github.com/qlik-trial/qlik-cli"
```
Comment (wennmo) [inline Formula/qlik-cli.rb]: Is this required?

File: publish.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	Name string `json"name"`
+	Tag  string
+}
+
+func main() {
+	token = os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		check(fmt.Errorf("GITHUB_TOKEN not set"))
+	}
+	base := "https://api.github.com/repos/qlik-trial/qlik-cli"
```
Comment (wennmo) [inline publish.go]: can we consume the base url to the private repo from an env variable instead?

File: Formula/qlik-cli.rb (hunk 1 above)
Comment (glooms) [reply Formula/qlik-cli.rb]: I think so!

File: publish.go (hunk 2 above)
Comment (glooms) [reply publish.go]: Sure thing

File: Formula/qlik-cli.rb (hunk 1 above)
Comment (wennmo) [reply Formula/qlik-cli.rb]: Ok, then I suggest we put the homebrew-taps repo as homepage since, the private repo will not be accessible externally.

Comment (glooms) [reply Formula/qlik-cli.rb]: It wasn't needed, turns out.

File: README.md [hunk 3]
```diff
@@ -11,8 +11,12 @@ To tap this tap just run
 ```
 brew tap qlik-oss/taps
 ```
-After the tap is tapped, you can easily install any formula present in this repo, e.g. corectl:
+After the tap is tapped, you can easily install any formula present in this repo, corectl or qlik-cli:
 ```
 brew install qlik-corectl
+brew install qlik-cli
```
Comment (wennmo) [inline README.md]: we should probably have this instruction in the private repo

**Review by wennmo — APPROVED** [review]
🎉

**Pull request merged**